app/assets/javascripts/discourse/services/theme-settings.js.es6

```diff
@@ -0,0 +1,20 @@
+export default Ember.Service.extend({
+  settings: null,
+
+  init() {
+    this._super(...arguments);
+    this.set("settings", {});
+  },
+
+  registerSettings(themeId, settingsObject) {
+    this.get("settings")[themeId] = settingsObject;
```

Comment:
If everything is accessed via the helpers, it's slightly faster to use plain old javascript objects. For example `this._settings = {}` in the constructor, and `this._settings[themeId] = settingsObject` here.

In general I recommend this unless something observes or computes a property based on the object, which doesn't seem to be the case in this interface.